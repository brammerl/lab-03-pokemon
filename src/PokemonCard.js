import React, { Component } from 'react'

export default class PokemonCard extends Component {
    render() {
        return (
           
            <li className='box'>
                <img src={this.props.pokemon.url_image} alt={this.props.pokemon.pokemon}/>
                <h1>{this.props.pokemon.pokemon}</h1>
                <ul>
                    <li>Type: {this.props.pokemon.type_1} {this.props.pokemon.type_2}</li>
                    <li>Attack: {this.props.pokemon.attack}</li>
                    <li>Defense: {this.props.pokemon.defense}</li>
                </ul>
            </li>
            
        )
    }
}
