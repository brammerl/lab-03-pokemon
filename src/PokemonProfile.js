import React, { Component } from 'react'

export default class PokemonProfile extends Component {
    render() {
        return (
            <div>
                <h1 id='pokemonName'>{this.props.name.pokemon}</h1>
                    <section id='pokemonBox'>
                        <img src={this.props.name.url_image} alt='Look! A cute pokemon!' id='profileImg'/>
                        <section id='information'>
                            <ul id='basicInfo'> 
                                <li><h2>Basic Information</h2></li>
                                <li>Generation: {this.props.name.generation_id}</li>
                                <li>Primary Type: {this.props.name.type_1}</li>
                                <li>Height: {this.props.name.height} meters</li>
                                <li>Weight: {this.props.name.weight} kg</li>
                                <li>Base XP: {this.props.name.base_experience}</li>
                                <li>HP: {this.props.name.hp}</li>
                                <li><a href={this.props.name.pokedex}>More Info</a></li>
                            </ul>
                            
                    </section>
                </section>
            </div>
        )
    }
}
