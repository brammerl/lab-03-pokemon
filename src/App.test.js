import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header.js';
import Selector from './Select.js';
import PokemonProfile from './PokemonProfile.js'
import PokemonCard from './PokemonCard.js'


const testObj = {
  "_id": "5cef3501ef6005a77cd4fd16",
  "pokemon": "venusaur",
  "id": 3,
  "species_id": 3,
  "height": 20,
  "weight": 1000,
  "base_experience": 236,
  "type_1": "grass",
  "type_2": "poison",
  "attack": 82,
  "defense": 83,
  "hp": 80,
  "special_attack": 100,
  "special_defense": 100,
  "speed": 80,
  "ability_1": "overgrow",
  "ability_2": "NA",
  "ability_hidden": "chlorophyll",
  "color_1": "#78C850",
  "color_2": "#A040A0",
  "color_f": "#81A763",
  "egg_group_1": "monster",
  "egg_group_2": "plant",
  "url_image": "http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png",
  "generation_id": 1,
  "evolves_from_species_id": "2",
  "evolution_chain_id": 1,
  "shape_id": 8,
  "shape": "quadruped",
  "pokebase": "venusaur",
  "pokedex": "http://www.pokemon.com/us/pokedex/venusaur"
  }

it('renders search bar', () => {
  const wrapper = shallow(<Selector />);

  expect(wrapper).toMatchSnapshot();
});

it ('renders the pokemon profile', () => {
  const wrapper = shallow(<PokemonProfile pokemon={testObj}/>)

  expect(wrapper).toMatchSnapshot();
})

it ('renders header', () => {
  const wrapper = shallow(<Header/>)

  expect(wrapper).toMatchSnapshot();
})

it ('renders pokemon general card', () => {
  const wrapper = shallow(<PokemonCard pokemon={testObj}/>)

  expect(wrapper).toMatchSnapshot();
})

