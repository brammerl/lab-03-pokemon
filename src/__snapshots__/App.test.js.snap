// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates profile2 1`] = `
<div>
  <h1
    id="pokemonName"
  >
    venusaur
  </h1>
  <section
    id="pokemonBox"
  >
    <img
      alt="Look! A cute pokemon!"
      id="profileImg"
      src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
    />
    <section
      id="information"
    >
      <ul
        id="basicInfo"
      >
        <li>
          <h2>
            Basic Information
          </h2>
        </li>
        <li>
          Generation: 
          1
        </li>
        <li>
          Primary Type: 
          grass
        </li>
        <li>
          Height: 
          20
           meters
        </li>
        <li>
          Weight: 
          1000
           kg
        </li>
        <li>
          Base XP: 
          236
        </li>
        <li>
          HP: 
          80
        </li>
        <li>
          <a
            href="http://www.pokemon.com/us/pokedex/venusaur"
          >
            More Info
          </a>
        </li>
      </ul>
    </section>
  </section>
</div>
`;

exports[`renders header 1`] = `
<header>
  <h1>
    Pokedex
  </h1>
</header>
`;

exports[`renders the drop down for type3 1`] = `
<div
  id="selector"
>
  <select>
    <option
      defaultValue={true}
      value=""
    >
      All
    </option>
    <option
      value="water"
    >
      Water
    </option>
    <option
      value="fire"
    >
      Fire
    </option>
    <option
      value="grass"
    >
      Grass
    </option>
  </select>
  <input
    placeholder="Enter Pokemone Name"
  />
  <button>
    Search
  </button>
</div>
`;
